### ISSUE TITLE
Trouble with Distribute challenge

### ISSUE THREAD

**Issue by EWCunha** [RareSkills/gas-puzzles#14]
I am having some trouble trying to optimize the Distribute contract. I believe that I need to do something about the transfer functions. I tried putting the 4 transfers in a for loop and also defining a local memory array that copies the contributors storage array, so that it does not get cold access for each element. But nothing of that worked. Could you give some tip that can help me solve this challenge? Thank you

**Comment by RareSkills** [comment]
Hi @EWCunha we can't put hints here, but this particular problem is indeed quite challenging. Note that if you copy the addresses from storage to memory, this will be an additional cost compared to just reading them from storage and doing a transfer.

Definitely try to think outside the box with this one!

**Comment by EWCunha** [comment]
Well, that doesn't help at all. I am honeslty out of ideas. I tried googling for gas efficient ways to send ETH to a batch of addresses, but didn't find any. I am also doing the Yul and Assembly Udemy course and tried some Yul code, but also didn't achieve the gas target. I tried storing the contract's balance to a variable, because it seems to me that reading the contract's balance costs a lot of gas, but on the test script, the balance is set artificially, without actually sending ETH (which seems like cheating to me), which does not update state variables.
